hr_contract_cae/models/hr_contract.py

```diff
@@ -1,12 +1,137 @@
+        inverse_name="contract_group_id",
+        string="Contracts",
+    )
+
+    def get_initial(self):
+        return self.contract_ids.search(
+            [("contract_group_id", "=", self.id)], limit=1
+        )
+
+    def get_parent(self, amendment_index):
+        return self.contract_ids.search(
+            [
+                ("contract_group_id", "=", self.id),
+                ("amendment_index", "<", amendment_index),
+            ],
+            order="amendment_index desc",
+            limit=1,
+        )
+
+    def get_child(self, amendment_index):
+        return self.contract_ids.search(
+            [
+                ("contract_group_id", "=", self.id),
+                ("amendment_index", ">", amendment_index),
+            ],
+            limit=1,
+        )
+
+    def get_latest(self):
+        return self.contract_ids.search(
+            [("contract_group_id", "=", self.id)],
+            order="amendment_index desc",
+            limit=1,
+        )
+
+
+class ContractType(models.Model):
+
+    _inherit = "hr.contract.type"
+
+    echelon = fields.Selection(
+        [("main", "Main"), ("amendment", "Amendment")],
+        string="Echelon",
+        default="main",
+    )
+    max_usage = fields.Integer(string="Maximum Usage", default=False)
 
 
 class Contract(models.Model):
     _inherit = "hr.contract"
 
-    reason = fields.Char(string="Reason for recourse to contract")
+    name = fields.Char(compute="_compute_name")
+    employee_id = fields.Many2one(required=True)
+    state = fields.Selection(copy=False)
+    state_admin = fields.Text(string="Administrative state", copy=False)
+    type_id = fields.Many2one(
+        string="Contract Type",
+        domain="[('echelon', '=', type_echelon)]",
+        default="",
+        copy=False,
+    )  # Todo: rename translation from "Catégorie de l'employé" to "Type de Contrat"
+    type_echelon = fields.Selection(
+        [("main", "Main"), ("amendment", "Amendment")],
+        default="main",
+        string="Contract Type Echelon",
+        copy=False,
+        compute="_compute_type_echelon",
+        store=True,
+    )  # Todo: add translation "Echelone du Type de Contrat"
+    type_count = fields.Integer(
+        string="Contract Type Count", compute="_compute_type_count"
+    )
+
+    amendment_index = fields.Integer(
+        string="Amendment Index",
+        help="'O' for main contract, '1' for it's first amendment, etc.",
+        default=0,
+        required=True,
+        readonly=True,
+    )
+    contract_group_id = fields.Many2one(
+        comodel_name="hr.contract.group", string="Contract Group"
+    )
+    contract_group_contract_ids = fields.One2many(
+        comodel_name="hr.contract", related="contract_group_id.contract_ids"
+    )
+    initial_contract_id = fields.Many2one(
+        comodel_name="hr.contract",
+        string="Initial Contract",
+        readonly=True,
+        compute="_compute_initial",
+    )
```

Comment:
Option A: Are all these fields needed on the contract after your refactoring?
Option B: Shouldn't they rather be related fields on contract_group_contract_ids?